bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -259,21 +391,545 @@ public void sendAutomowerCommand(AutomowerCommand command, long commandDurationM
+        String id = automowerId.get();
+        try {
+            AutomowerBridge automowerBridge = getAutomowerBridge();
+            if (automowerBridge != null) {
+                automowerBridge.sendAutomowerResetCuttingBladeUsageTime(id);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/conf-error-no-bridge");
+            }
+        } catch (AutomowerCommunicationException e) {
+            logger.warn("Unable to send ResetCuttingBladeUsageTime to automower: {}, Error: {}", id, e.getMessage());
+        }
         updateAutomowerState();
     }
 
     private String restrictedState(RestrictedReason reason) {
         return "RESTRICTED_" + reason.name();
     }
 
-    private void updateChannelState(@Nullable Mower mower) {
+    private @Nullable String getErrorMessage(int errorCode) {
+        return ERROR.get(errorCode);
+    }
+
+    private @Nullable WorkArea getWorkAreaById(@Nullable Mower mower, long workAreaId) {
+        if (mower != null) {
+            List<WorkArea> workAreas = mower.getAttributes().getWorkAreas();
+            for (WorkArea workArea : workAreas) {
+                if (workArea.getWorkAreaId() == workAreaId) {
+                    return workArea;
+                }
+            }
+        }
+        return null;
+    }
+
+    private void updateChannelState(@Nullable Mower mower, @Nullable MowerMessages mowerMessages) {
         if (isValidResult(mower)) {
+            Capabilities capabilities = mower.getAttributes().getCapabilities();
+
+            /*
+             * Update channels based on the received data
+             */
+            // create a copy of the present channels
+            channelAdd.clear();
+            for (Channel channel : thing.getChannels()) {
+                channelAdd.add(channel);
+            }
+            channelRemove.clear();
+
+            if (capabilities.hasWorkAreas()) {
+                createChannel(CHANNEL_STATUS_WORK_AREA_ID, CHANNEL_TYPE_STATUS_WORK_AREA_ID, "Number");
+                createChannel(CHANNEL_STATUS_WORK_AREA, CHANNEL_TYPE_STATUS_WORK_AREA, "String");
+            } else {
+                removeChannel(CHANNEL_STATUS_WORK_AREA_ID);
+                removeChannel(CHANNEL_STATUS_WORK_AREA);
+            }
+            if (capabilities.canConfirmError()) {
+                createChannel(CHANNEL_STATUS_ERROR_CONFIRMABLE, CHANNEL_TYPE_STATUS_ERROR_CONFIRMABLE, "Switch");
+            } else {
+                removeChannel(CHANNEL_STATUS_ERROR_CONFIRMABLE);
+            }
+
+            if (capabilities.hasHeadlights()) {
+                createChannel(CHANNEL_SETTING_HEADLIGHT_MODE, CHANNEL_TYPE_SETTING_HEADLIGHT_MODE, "String");
+            } else {
+                removeChannel(CHANNEL_SETTING_HEADLIGHT_MODE);
+            }
+
+            if (capabilities.hasPosition()) {
+                // if (mower.getAttributes().getLastPosition() != null) {
```

Comment:
Can this comment be removed?

Reply:
yes, as the original code was replaced by the line above.